src/oal_agent/llm/provider.py

```
@@ -43,27 +49,59 @@ def __init__(self, api_key: str):
+                result = await asyncio.wait_for(self._call_openai_api(prompt, **kwargs), timeout=timeout)
+                return result
+            except asyncio.TimeoutError as e:
+                if attempt < retry_attempts - 1:
+                    backoff_time = 2**attempt + random.uniform(0, 1)
+                    print(f"OpenAI API call timed out. Retrying in {backoff_time:.2f} seconds...")
+                    await asyncio.sleep(backoff_time)
+                else:
+                    raise LLMTimeoutError("OpenAI API call timed out after multiple retries.") from e
+            except Exception as e:
+                if attempt < retry_attempts - 1:
+                    backoff_time = 2**attempt + random.uniform(0, 1)
```

Ruff (0.14.3)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)